Library/Homebrew/diagnostic.rb

```diff
@@ -100,7 +100,7 @@ def check_for_installed_developer_tools
 
       # See https://github.com/Homebrew/legacy-homebrew/pull/9986
       def check_path_for_trailing_slashes
-        all_paths = ENV["PATH"].split(File::PATH_SEPARATOR)
+        all_paths = PATH.new(ENV["PATH"]).to_a
```

Comment:
Not sure I understand why you'd want to create it and make it an array immediately. Would be nice to perhaps have a solution like @alyssais's or a new method that takes a string and returns an array.